lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -981,8 +986,13 @@ impl Generator {
             })
             .unwrap();
 
-            // Some items might still be pending a resolve flush. Do it now before we start the next stage.
-            self.force_flush_resolved(is_tiling_mode);
+            {
+                // append all resolved threads to resolved
+                let mut resolved = self.resolved.write().unwrap();
+                for thread_resolved in pixels_resolved_this_stage {
+                    resolved.extend(thread_resolved.lock().unwrap().iter());
```

Comment:
Might be a bit clearer like this:

```suggestion
                    resolved.append(&mut thread_resolved.into_inner().unwrap());
```